### PR TITLE
Add users to system

### PR DESCRIPTION
**Pull request by lior1-guz**
1. This is the implementation of the classes in "user" package including:
1) User
2) Customer
3) Business customer
4) HR manager
5) CEO 
6) Branch manager
7) Supplier

***Each class includes constructor + toString + attributes***

2. In this pull request I fixed all the dependencies between our project. 

***Notice***
The attributes are bit different from our class diagram -> needs to be carefully reviewed

### CODE REVIEW

**Review by orimalka — APPROVED** [review]

**Review by orimalka — APPROVED** [review]

**Comment by orimalka** [comment]
Please make sure guys that we don't have more then two lines between parameters and method to method.

File: G3Common/src/users/BranchManager.java [hunk 1]
```diff
@@ -52,4 +51,11 @@ public void setBranchName(Branch branchName) {
 		this.branchName = branchName;
 	}
 
+	/**
+	 * This is the toString for this class
+	 */
+	@Override
+	public String toString() {
+		return "Branch manager: "  + userFirstName + " " + userLastName + ", user ID = " + userId;
```
Comment (orimalka) [inline G3Common/src/users/BranchManager.java]: I am not sure that we need to enter here his ID, I would enter the branch name instead. 
I suggest: return branchName + "manager: " + userFirstName + " " + userLastName;

File: G3Common/src/users/BusinessCustomer.java [hunk 2]
```diff
@@ -2,83 +2,80 @@
 
 /**
  * 
- * @author Lior, Guzovsky.
- * Class description: 
- * This class is derived class from Customer class
- *  which defines the main attributes and functionalities
- * of a Business customer in our system.
+ * @author Lior, Guzovsky. Class description: This class is derived class from
```
Comment (orimalka) [inline G3Common/src/users/BusinessCustomer.java]: please add line after your name

File: G3Common/src/users/BusinessCustomer.java [hunk 3]
```diff
@@ -2,83 +2,80 @@
...
 	 * This is the customers position in the company.
 	 */
 	protected String businessCustomerPosition;
-	
-	
+
 	/**
-	 * Each Business customer has a budget max
-	 * that he can use
+	 * Each Business customer has a budget max that he can use
```
Comment (orimalka) [inline G3Common/src/users/BusinessCustomer.java]: Each business customer has a budget that he cannot exceed.

File: G3Common/src/users/BusinessCustomer.java [hunk 4]
```diff
@@ -104,4 +101,20 @@ public void setBudgetMaxAmount(int budgetMaxAmount) {
...
+	public void setBusinessW4cCode(int businessW4cCodeNumber) {
+		this.businessW4cCodeNumber = businessW4cCodeNumber;
+	}
+	
+	/**
+	 * This is the toString for this class
+	 */
+	@Override
+	public String toString() {
+		return "Business Customer: "  + userFirstName + " " + userLastName + ", user ID = " + userId;
```
Comment (orimalka) [inline G3Common/src/users/BusinessCustomer.java]: Same as before, remove user id and put company name instead

File: G3Common/src/users/CeoBiteMe.java [hunk 5]
```diff
@@ -10,27 +10,32 @@
...
 		
 	}
 
-	
+	/**
+	 * This is the toString for this class
+	 */
+	@Override
+	public String toString() {
+		return "CeoBiteMe: " + userFirstName + " " + userLastName + " CEO ID: " + userId;
```
Comment (orimalka) [inline G3Common/src/users/CeoBiteMe.java]: same, if you want to keep the userId its up to you since you did it for all of them.

**Review by orimalka — CHANGES_REQUESTED** [review]
Fix toStrings as we talked.. good job

**Review by orimalka — APPROVED** [review]

File: G3Common/src/users/BusinessCustomer.java (hunk 2 above)
Comment (lior1-guz) [reply G3Common/src/users/BusinessCustomer.java]: fixed

File: G3Common/src/users/BranchManager.java (hunk 1 above)
Comment (lior1-guz) [reply G3Common/src/users/BranchManager.java]: fixed

File: G3Common/src/users/BusinessCustomer.java (hunk 3 above)
Comment (lior1-guz) [reply G3Common/src/users/BusinessCustomer.java]: fixed

File: G3Common/src/users/BusinessCustomer.java (hunk 4 above)
Comment (lior1-guz) [reply G3Common/src/users/BusinessCustomer.java]: fixed

File: G3Common/src/users/CeoBiteMe.java (hunk 5 above)
Comment (lior1-guz) [reply G3Common/src/users/CeoBiteMe.java]: fixed

**Review by orimalka — APPROVED** [review]

**Pull request merged**